File: app/src/main/java/jp/superwooo/chaipon/lighttimeswitcher/SystemScreenOffTimeoutAccessor.java
package jp.superwooo.chaipon.lighttimeswitcher;

import android.content.ContentResolver;
import android.content.Context;
import android.provider.Settings;

public class SystemScreenOffTimeoutAccessor {
    private Context mContext;
    public static SystemScreenOffTimeoutAccessor create(Context context){
        return new SystemScreenOffTimeoutAccessor(context);
    }
    private SystemScreenOffTimeoutAccessor(Context context){
        mContext = context;
    }
    public TimeDurationValue read(){
        ContentResolver cr = mContext.getContentResolver();
        try {
            int timeOut = Settings.System.getInt(cr, Settings.System.SCREEN_OFF_TIMEOUT) / 1000;
            return new TimeDurationValue(timeOut, SettingsActivity.LimitTime);
        } catch (Settings.SettingNotFoundException e) {
            e.printStackTrace();
            return new TimeDurationValue(15, SettingsActivity.LimitTime);
        }
    }
    public void write(TimeDurationValue timeout){
        Settings.System.putInt(mContext.getContentResolver(), Settings.System.SCREEN_OFF_TIMEOUT, timeout.milliSecond());
    }
}

File: app/src/main/java/jp/superwooo/chaipon/lighttimeswitcher/SystemScreenOffTimeoutAccessor.kt
package jp.superwooo.chaipon.lighttimeswitcher

import android.content.Context
import android.provider.Settings
import android.provider.Settings.SettingNotFoundException

class SystemScreenOffTimeoutAccessor private constructor(private val context: Context) {
    fun read(): TimeDurationValue {
        val cr = context.contentResolver
        try {
            val timeOut = Settings.System.getInt(cr, Settings.System.SCREEN_OFF_TIMEOUT) / 1000
            return TimeDurationValue(timeOut, SettingsActivity.Companion.LimitTime)
        } catch (e: SettingNotFoundException) {
            e.printStackTrace()
            return TimeDurationValue(15, SettingsActivity.Companion.LimitTime)
        }
    }

    fun write(timeout: TimeDurationValue?) {
        Settings.System.putInt(
            context.contentResolver,
            Settings.System.SCREEN_OFF_TIMEOUT,
            timeout!!.milliSecond()
        )
    }

    companion object {
        fun create(context: Context): SystemScreenOffTimeoutAccessor {
            return SystemScreenOffTimeoutAccessor(context)
        }
    }
}
